runtimes/core/src/api/cors/mod.rs

```diff
@@ -62,21 +75,45 @@ pub fn config(cfg: &pb::gateway::Cors, meta: MetaHeaders) -> anyhow::Result<Cors
             }
             _ => OriginSet::Some(vec![]),
         };
-        let without_creds = OriginSet::new(
-            cfg.allowed_origins_without_credentials
-                .clone()
-                .unwrap_or_default()
-                .allowed_origins,
-        );
+        let without_creds = {
+            if let Some(CorsAllowedOrigins { allowed_origins }) =
+                &cfg.allowed_origins_without_credentials
+            {
+                OriginSet::new(allowed_origins.to_vec())
+            } else {
+                OriginSet::All
+            }
+        };
+
+        let request_has_creds = |req: &axum::http::request::Parts| -> bool {
+            if req.headers.contains_key(AUTHORIZATION) || req.headers.contains_key(COOKIE) {
+                return true;
+            }
+
+            if req.method == http::method::Method::OPTIONS {
+                return req
+                    .headers
+                    .get_all(ACCESS_CONTROL_REQUEST_HEADERS)
+                    .iter()
+                    .any(|val| {
+                        val.to_str()
+                            .map(|val| {
+                                val.split(",")
+                                    .map(|val| val.trim())
```

Comment:
not sure if the trim is actually needed, but I couldn't find any reference in the docs that says explicitly that it is invalid to have a space after the comma